spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/cql/QueryOptionsUtil.java

```diff
@@ -106,6 +106,9 @@ public static Insert addWriteOptions(Insert insert, WriteOptions writeOptions) {
 		if (!writeOptions.getTtl().isNegative()) {
 			insert.using(QueryBuilder.ttl(Math.toIntExact(writeOptions.getTtl().getSeconds())));
 		}
+		if (!(writeOptions.getTimestamp() < 0)) {
```

Comment:
`WriteOptions` (and `QueryOptions` in general) uses nullable properties to indicate presence/absence of options.